File: src/app/Models/WorkspaceProfile.cs
namespace CluckIn.App.Models;

public sealed record WorkspaceProfile
{
    public required string Id { get; init; }
    public required string Name { get; init; }
    public IReadOnlyList<string> AllowedApplications { get; init; } = [];
    public IReadOnlyList<string> AllowedDomains { get; init; } = [];
    public IReadOnlyList<string> AllowedWindowKeywords { get; init; } = [];
    public IReadOnlyList<string> BlockedApplications { get; init; } = [];
    public IReadOnlyList<string> BlockedWindowKeywords { get; init; } = [];
}

File: src/web/src/models/WorkspaceProfile.ts
export type WorkspaceProfile = {
  id: string;
  name: string;
  allowedApplications: string[];
  allowedWindowKeywords: string[];
  blockedApplications: string[];
  blockedWindowKeywords: string[];
};

export type RuleCategory = Exclude<keyof WorkspaceProfile, 'id' | 'name'>;
